tests/lib/rules/array-bracket-spacing.js

```diff
@@ -159,8 +174,11 @@ ruleTester.run("array-bracket-spacing", rule, {
         { code: "var foo = [1, {'bar': 'baz'}, 5];", options: ["never"] },
         { code: "var foo = [{'bar': 'baz'}, 1,  5];", options: ["never"] },
         { code: "var foo = [1, 5, {'bar': 'baz'}];", options: ["never"] },
-        { code: "var obj = {'foo': [1, 2]}", options: ["never"] }
+        { code: "var obj = {'foo': [1, 2]}", options: ["never"] },
 
+        // destructuring with type annotation
+        { code: "([ 1, 1 ]: Array<any>)", options: ["always"], parserOptions: { ecmaVersion: 6 }, parser: parser("flow-destructuring-1") },
+        { code: "([1, 1]: Array< any >)", options: ["never"], parserOptions: { ecmaVersion: 6 }, parser: parser("flow-destructuring-2") },
```

Comment:
I found it, those tests are [Typecasts syntax](https://flowtype.org/docs/syntax.html#typecasts).
Those `ArrayExpression` nodes don't have `typeAnnotation`s, so those tests are not proper.